pkg/preflight/preflight.go

```diff
@@ -141,8 +142,23 @@ func Run(appID string, appSlug string, sequence int64, isAirgap bool, archiveDir
 		var preflightErr error
 		defer func() {
 			if preflightErr != nil {
-				err := setPreflightResult(appID, sequence, &types.PreflightResults{}, preflightErr)
+				preflightResults := &types.PreflightResults{
```

Comment:
how about combining this with the `setResults` function below? it can also be defined outside of entire function to make it smaller. same for setProgress.